src/ace_time/hw/StmRtc.cpp

```diff
@@ -17,13 +17,18 @@ using ace_common::decToBcd;
 namespace ace_time {
 namespace hw {
 
-StmRtc::StmRtc() {
+StmRtc::StmRtc() {}
 
+bool StmRtc::begin(const sourceClock_t clockSource, const hourFormat_t hourFormat) {
+  
   mRtc = &STM32RTC::getInstance();
   if (mRtc) {
```

Comment:
Curious: do we really need this check? Because it looks like `STM32RTC::getInstance()` returns a reference to a `static STM32RTC instance`. A reference can never be null, since taking a reference of `nullptr` is Undefined Behavior (https://stackoverflow.com/questions/6716091). So taking a pointer of that reference will never produce a `nullptr`. There is no `operator STM32RTC::bool()` that I can see, so this will never be false.